components/server/src/api/auth-provider-service-api.ts

```diff
@@ -23,21 +23,24 @@ import {
     DeleteAuthProviderResponse,
 } from "@gitpod/public-api/lib/gitpod/v1/authprovider_pb";
 import { AuthProviderService } from "../auth/auth-provider-service";
-import { AuthProviderEntry, AuthProviderInfo } from "@gitpod/gitpod-protocol";
+import { AuthProviderEntry, AuthProviderInfo, User } from "@gitpod/gitpod-protocol";
 import { Unauthenticated } from "./unauthenticated";
 import { validate as uuidValidate } from "uuid";
 import { selectPage } from "./pagination";
+import { ctxUserId } from "../util/request-context";
+import { UserService } from "../user/user-service";
 
 @injectable()
 export class AuthProviderServiceAPI implements ServiceImpl<typeof AuthProviderServiceInterface> {
     constructor(
         @inject(PublicAPIConverter) private readonly apiConverter: PublicAPIConverter,
         @inject(AuthProviderService) private readonly authProviderService: AuthProviderService,
+        @inject(UserService) private readonly userService: UserService,
     ) {}
 
     async createAuthProvider(
         request: CreateAuthProviderRequest,
-        context: HandlerContext,
+        _: HandlerContext,
     ): Promise<CreateAuthProviderResponse> {
         const ownerId = request.owner.case === "ownerId" ? request.owner.value : "";
```

Comment:
there is it used?

Reply:
What do you mean? :thinking: 
I think there is no use-case for accessing HandlerContext in actual service implementations, no?

Reply:
I mean ownerId is not used in the code below, instead we use ctxUserId

Reply:
Hm, what I did was just "replace context.user.id with ctxUserId()".

But I think you are right: ctxUserId is correct for authroization (first argument), but ownerId should be used in the shape - at least in the path where we validate it.

@AlexTugarev Could you review that section as well? Would be ideal if the two of us could sync, I could be missing sth here.